src/test/scala/com/mozilla/telemetry/stats/MannWhitneyU.scala

```diff
@@ -0,0 +1,139 @@
+        Map(1 -> 3.0, 2 -> 7.5, 3 -> 11.0, 4 -> 13.5, 5 -> 15.0))
+  }
+
+  "mwu with normal distribution" can "be calculated" in {
+    val norm1 = Map(0 -> 42, 1 -> 60, 2 -> 92, 3 -> 100, 4 -> 109, 5 -> 120, 6 -> 128, 7 -> 92, 8 -> 76, 9 -> 42, 10 -> 50, 11 -> 11, 12 -> 11, 13 -> 1, 14 -> 3, 15 -> 3, 16 -> 2, -1 -> 35, -6 -> 1, -5 -> 1, -4 -> 2, -3 -> 4, -2 -> 15)
+    val norm2 = Map(0 -> 6, 1 -> 23, 2 -> 39, 3 -> 91, 4 -> 119, 5 -> 142, 6 -> 157, 7 -> 139, 8 -> 122, 9 -> 80, 10 -> 45, 11 -> 20, 12 -> 10, 13 -> 4, 14 -> 1, -1 -> 2)
+    var res = MannWhitneyU.mwu(norm1, norm2)
+    res._1 should equal (381391.0)
+    res._2 should equal (1.2950845524396078e-20 +- 0.000001)
+    res = MannWhitneyU.mwu(norm1, norm2, false)
+    res._1 should equal (381391.0)
+    res._2 should equal (1.2946137151181732e-20 +- 0.000001)
+  }
+
+  "mwu with uniform distribution" can "be calculated" in {
+    val uni1 = Map(1 -> 25, 2 -> 23, 3 -> 17, 4 -> 25, 5 -> 18, 6 -> 15, 7 -> 18, 8 -> 21, 9 -> 20, 10 -> 22, 11 -> 25, 12 -> 18, 13 -> 21, 14 -> 17, 15 -> 23, 16 -> 16, 17 -> 14, 18 -> 20, 19 -> 18, 20 -> 24, 21 -> 19, 22 -> 26, 23 -> 18, 24 -> 20, 25 -> 19, 26 -> 23, 27 -> 16, 28 -> 24, 29 -> 16, 30 -> 21, 31 -> 13, 32 -> 16, 33 -> 22, 34 -> 27, 35 -> 22, 36 -> 14, 37 -> 29, 38 -> 22, 39 -> 10, 40 -> 23, 41 -> 22, 42 -> 26, 43 -> 19, 44 -> 35, 45 -> 17, 46 -> 23, 47 -> 18, 48 -> 16, 49 -> 24)
+    val uni2 = Map(10 -> 15, 11 -> 18, 12 -> 17, 13 -> 16, 14 -> 18, 15 -> 23, 16 -> 24, 17 -> 13, 18 -> 15, 19 -> 12, 20 -> 14, 21 -> 23, 22 -> 14, 23 -> 11, 24 -> 15, 25 -> 18, 26 -> 22, 27 -> 15, 28 -> 19, 29 -> 14, 30 -> 22, 31 -> 26, 32 -> 18, 33 -> 9, 34 -> 25, 35 -> 19, 36 -> 25, 37 -> 17, 38 -> 14, 39 -> 23, 40 -> 12, 41 -> 10, 42 -> 20, 43 -> 25, 44 -> 17, 45 -> 26, 46 -> 20, 47 -> 18, 48 -> 23, 49 -> 19, 50 -> 16, 51 -> 17, 52 -> 21, 53 -> 20, 54 -> 12, 55 -> 14, 56 -> 17, 57 -> 28, 58 -> 16, 59 -> 15)
+    var res = MannWhitneyU.mwu(uni1, uni2)
+    res._1 should equal (290777.0)
+    res._2 should equal (7.03102430020267e-41 +- 0.000001)
+    res = MannWhitneyU.mwu(uni1, uni2, false)
+    res._1 should equal (290777.0)
+    res._2 should equal (7.027076082562216e-41 +- 0.000001)
+  }
+
+  "mwu with skewed distribution" can "be calculated" in {
+    val skew1 = Map(0 -> 383, 1 -> 465, 2 -> 137, 3 -> 14, 4 -> 1)
+    val skew2 = Map(0 -> 362, 1 -> 416, 2 -> 107, 3 -> 14, 5 -> 1)
+    var res = MannWhitneyU.mwu(skew1, skew2)
+    res._1 should equal (438324.5)
+    res._2 should equal (0.14272273391953955 +- 0.000001)
+    res = MannWhitneyU.mwu(skew1, skew2, false)
+    res._1 should equal (438324.5)
+    res._2 should equal (0.14271241840387777 +- 0.000001)
+  }
+}
+
+
+/*
+ * The above MWU test values were pulled from the Python moztelemetry project,
+ * which tests against scipy.
+ *
+ * Below is sample code to create the input distributions and resulting values.
+ *
+ *
+ * NORMAL DISTRIBUTION
+ *
+ * >>> import numpy
```

Comment:
Maybe it's better to add a python script that can generate these values, which is easier to read and use than a comment.